[Cocoanautz] ip

cocoanautz opened the pull request:
# Drew, **_The_** chat-bot of all time.
> "He who does not remember his tasks, will not do them." - Sun Tzu

Do you want to be:
- [ ] Productive
- [ ] Up to date
- [ ] Great

If you answered **YES** to all the above, then Drew is **_The_** buddy for you!

All you need to do to achieve greatness is:
1. Download the code from [here](https://github.com/cocoanautz/ip).
2. Start up the chat-bot.
3. Key in your tasks.

Features include:
- Cool pal to talk to 😎 
- No nonsense. Only do what you need to do. 💯 

Look at how cool this response in the `executeCommand` method is:
``` java
int firstBackslashIndex = input.indexOf("/by");
if (firstBackslashIndex == -1) {
    throw new IllegalArgumentException("Incorrect input. Ensure that date begins with /by");
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

cocoanautz committed:
feat: chatbot greeting

cocoanautz committed:
Added greetings

cocoanautz committed:
Merge commit '29d602cd98690c5dd5366a870228e8ca03a2d172'

# Conflicts:
#	src/main/java/Duke.java

cocoanautz committed:
Added echo and exit function

cocoanautz committed:
Added Add, List functions

cocoanautz committed:
Added Task Completion Tracking

cocoanautz committed:
Added different types of tasks

cocoanautz committed:
Added tests

cocoanautz committed:
Added exception handling for: Empty arguments, illegal arguments

cocoanautz committed:
Added delete command

cocoanautz committed:
Add save function

cocoanautz committed:
Merge branch 'branch-Level-7'

cocoanautz committed:
Change date datatype

Date datatype in Deadline and Event objects have been updated from String to LocalDate to support more date related functions.

cocoanautz committed:
Merge branch 'branch-level-8'

cocoanautz committed:
Refactor storage process

Moved the save and load functions out of the main method to decouple storage functionality from main logic. Storage is now accessed with filePath and Arraylist.

cocoanautz committed:
Refactor Drew main method.

Refactored UI, Storage, Parsing and List functionality into respective packages and classes.

maze508 commented on src/main/java/drew/tasktypes/Task.java:
```diff
@@ -0,0 +1,32 @@
+        this.isDone = false;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Do not repeat in comments information that is already obvious from the code

maze508 commented on src/main/java/Drew.java:
```diff
@@ -0,0 +1,309 @@
+                throw new InsufficientArgumentsException("'Mark index' cannot be empty");
+            }
+            break;
+        //valid integer will be checked later
```
Comments should target other programmers reading the code and not only make sense to the person who wrote it

maze508 commented on src/main/java/Drew.java:
```diff
@@ -0,0 +1,309 @@
+     * @param input String containing full user input.
+     * @return Command specified by user input.
+     */
+    public static Command checkCommandIdentity(String input) throws UnknownCommandException,
```
Avoid complicated expressions and definitions. Your method `checkCommandIdentity` gets the userCommand by checking for string equivalence, then uses the userCommand to perform a conditional switch case check. This can be combined into 1 and reduce redundant checkings and simplify your method to make it more easily readble

maze508 commented on src/main/java/Drew.java:
```diff
@@ -0,0 +1,309 @@
+        }
+    }
+
+    public static String executeCommand(ArrayList<Task> ls, String input) {
```
Avoid Long Methods, you can consider calling another method to process your different userCommands

maze508 reviewed (commented):
LGTM. There are only some minor code quality issues that you can easily fix and missing javadocs for some of your classes and methods. Well Done!

cocoanautz committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2324S2/ip

cocoanautz committed:
Added unit tests for Parser.java and Storage.Java

Unit tests currently cover Parser.checkCommandIdentity() and Storage.parseSave().

For checkCommandIdentity(), it tests whether the parser recognizes the identity of valid commands.

For parseSave(), it tests for corruption in status symbol, argument count and date format issues.

cocoanautz committed:
Updated build.gradle to support Jar.

cocoanautz committed:
Add JavaDoc for most public methods.

cocoanautz committed:
Align all files to coding standard.

cocoanautz committed:
Merge branch 'branch-A-CodingStandard'

cocoanautz committed:
Merge branch 'branch-A-JavaDoc'

cocoanautz committed:
Add find command

cocoanautz committed:
Merge branch 'branch-level-9'

yongkotaro commented on src/main/java/drew/ui/Parser.java:
```diff
@@ -0,0 +1,234 @@
+
+        int inputLength = input.length();
+        Command userCommand;
+        if (inputLength == 3 && input.substring(0, 3).equalsIgnoreCase("bye")) {
```
Maybe could name input.substring(start, end) as "String command".

yongkotaro commented on src/main/java/drew/ui/Ui.java:
```diff
@@ -0,0 +1,48 @@
+     * The first message to be printed after load is complete.
+     */
+    public void greetUser() {
+        System.out.println(DELIMITER);
+        System.out.println("Hello! I'm drew.Drew");
+        System.out.println("What can I do for you?");
+        System.out.println(DELIMITER);
```
Maybe could combine them into a constant String GREETING

yongkotaro commented on src/main/java/drew/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
```
Could add JavaDocs, minor issue

yongkotaro commented on src/main/java/drew/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    public abstract String toSaveFormatString();
```
Maybe could add a description.

yongkotaro reviewed (commented):
Overall, the code followed the coding standard well with appropriate naming and statements. JavaDocs were missing for only 1-2 methods.

cocoanautz committed:
Add checkStyle to gradle.

cocoanautz committed:
Refactor command class

Separated command execution logic from parser. Parser only checks for command identity now.

cocoanautz committed:
Refactor command identity check into each command class.

cocoanautz committed:
Merge branch 'refactor-commands'

cocoanautz committed:
Add GUI to the chatbot.

cocoanautz committed:
Merge branch 'branch-level-10'

cocoanautz committed:
Drew.java: Clean unused code.

cocoanautz committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	src/main/java/drew/Drew.java
#	src/main/java/drew/ui/Parser.java

cocoanautz committed:
Update files to conform to checkstyle.

Most of the classes and methods minor checkstyle issues related to:
1. Class JavaDocs
2. Import statement ordering
3. Coding standards.

Fixing these issues will increase code readability, as well as
ensure that styles are consistent across all files.

cocoanautz committed:
Fix build.gradle

cocoanautz committed:
Add asserts in commands

Asserts were added into some commands to ensure that variables
are within the expected values during runtime.

These changes would increase the reliability of the program.

cocoanautz committed:
Merge pull request #2 from cocoanautz/branch-A-Assertions

Add asserts in commands

cocoanautz committed:
Add extension

The following extension have been added:
1. B-Reminders
2. C-DetectDuplicates

Reminders:
The chatbot would remind the user of any upcoming deadlines
whenever it starts.

DetectDuplicates:
The chatbot would prevent the user from creating tasks that already
exist.

cocoanautz committed:
Merge branch 'branch-Extension'

cocoanautz committed:
Update user guide

cocoanautz committed:
Merge branch 'branch-A-UserGuide'

cocoanautz committed:
Update parser tests after refactor

cocoanautz committed:
Merge branch 'branch-tests'

cocoanautz committed:
Refactor source code

cocoanautz committed:
Remove unused imports

cocoanautz committed:
Update User Guide

cocoanautz committed:
Merge branch 'branch-A-UserGuide'

cocoanautz committed:
Update dialog box size

Previously, Dialog boxes would terminate after 4 lines.
This commits removed max dialog box size.
